src/main/java/org/broadinstitute/hellbender/tools/spark/sv/evidence/FermiLiteAssemblyHandler.java

```diff
@@ -24,51 +26,509 @@
             }
             catch ( final IOException ioe ) {
                 throw new GATKException("Can't write "+gfaName, ioe);
             }
         }
-        final List<byte[]> tigSeqs =
-                assembly.getContigs().stream()
-                        .map(FermiLiteAssembly.Contig::getSequence)
-                        .collect(SVUtils.arrayListCollector(assembly.getNContigs()));
+
+        // align the assembled contigs to the genomic reference
         try ( final BwaMemAligner aligner = new BwaMemAligner(BwaMemIndexCache.getInstance(alignerIndexFile)) ) {
             aligner.setIntraCtgOptions();
-            final List<List<BwaMemAlignment>> alignments = aligner.alignSeqs(tigSeqs);
+            final List<byte[]> sequences =
+                    assembly.getContigs().stream()
+                            .map(Contig::getSequence)
+                            .collect(SVUtils.arrayListCollector(assembly.getNContigs()));
+            final List<List<BwaMemAlignment>> alignments = aligner.alignSeqs(sequences);
             return new AlignedAssemblyOrExcuse(intervalID, assembly, secondsInAssembly, alignments);
         }
     }
+
+    @VisibleForTesting
+    static FermiLiteAssembly reviseAssembly( final FermiLiteAssembly initialAssembly,
+                                             final boolean removeShadowedContigs,
+                                             final boolean expandAssemblyGraph ) {
+        final FermiLiteAssembly unshadowedAssembly =
+                removeShadowedContigs ? removeShadowedContigs(initialAssembly) : initialAssembly;
+        return expandAssemblyGraph ? expandAssemblyGraph(removeUnbranchedConnections(unshadowedAssembly)) : unshadowedAssembly;
+    }
+
+    @VisibleForTesting
+    static FermiLiteAssembly removeShadowedContigs( final FermiLiteAssembly assembly ) {
+        final int kmerSize = 31;
+        final double maxMismatchRate = .05;
+
+        // make a map of assembled kmers
+        final int capacity = assembly.getContigs().stream().mapToInt(tig -> tig.getSequence().length - kmerSize + 1).sum();
+        final HopscotchMultiMap<SVKmerShort, ContigLocation, KmerLocation> kmerMap = new HopscotchMultiMap<>(capacity);
+        assembly.getContigs().forEach(tig -> {
+            int contigOffset = 0;
+            final Iterator<SVKmer> contigItr = new SVKmerizer(tig.getSequence(), kmerSize, new SVKmerShort());
+            while ( contigItr.hasNext() ) {
+                final SVKmerShort kmer = (SVKmerShort)contigItr.next();
+                final SVKmerShort canonicalKmer = kmer.canonical(kmerSize);
+                final ContigLocation location = new ContigLocation(tig, contigOffset++, kmer.equals(canonicalKmer));
+                kmerMap.add(new KmerLocation(canonicalKmer, location));
+            }
+        });
+
+        // remove contigs that vary by a small number of SNPs from a sub-sequence of another contig
+        final Set<Contig> contigsToRemove = new HashSet<>();
```

Comment:
Assumption is correct, and the test on 159-160 is checking that.  It will be deterministic, as long as the order of contigs returned by FermiLite is deterministic:  we'll kill the first one.  I believe that FermiLite is deterministic, but I don't know for sure.